easy_rec/python/layers/seq_input_layer.py

```diff
@@ -22,43 +24,60 @@ def __init__(self, feature_configs, feature_groups_config):
 
   def __call__(self, features, group_name):
     feature_column_dict = self._fc_parser.deep_columns
+    feature_column_dict.update(self._fc_parser.sequence_columns)
 
     builder = feature_column._LazyBuilder(features)
-    key_features = []
-    hist_emb_features = []
-    hist_seqlen_features = []
+
     feature_dict = self._feature_groups_config[group_name]
+    seq_att_map = feature_dict.seq_att_map
+    keys = seq_att_map.key
+    hist_seqs = seq_att_map.hist_seq
+    tf_summary = feature_dict.tf_summary
+    if tf_summary:
+      logging.info('Write sequence feature to tensorflow summary.')
+
+    def _seq_embed_summary_name(input_name):
+      input_name = input_name.split(':')[0]
+      input_name = input_name.split('/')[:2]
+      return 'sequence_feature/' + '/'.join(input_name)
+
+    key_tensors = []
+    for key in keys:
+      qfc = feature_column_dict[key]
+      with tf.variable_scope(qfc._var_scope_name):
+        key_tensors.append(feature_column_dict[key]._get_dense_tensor(builder))
+    if tf_summary:
+      for key_tensor in key_tensors:
+        tf.summary.histogram(
+            _seq_embed_summary_name(key_tensor.name), key_tensor)
 
-    for x in feature_dict.seq_att_map:
-      with tf.variable_scope('emd_' + x.key, reuse=tf.AUTO_REUSE):
-        key_tensor = feature_column_dict[x.key]._get_dense_tensor(builder)
-      with tf.variable_scope('hist_emd_' + x.hist_seq, reuse=tf.AUTO_REUSE):
-        hist_embedding, hist_seq_len = feature_column_dict[
-            x.hist_seq]._get_sequence_dense_tensor(builder)
-
-      key_features.append(key_tensor)
-      hist_emb_features.append(hist_embedding)
-      hist_seqlen_features.append(hist_seq_len)
-
-      # din_features.append({'key': key_tensor, 'hist_seq_emb': hist_embedding, 'hist_seq_len': hist_seq_len})
-    if len(feature_dict.seq_att_map) > 1:
-      features = {
-          'key': tf.concat(key_features, axis=1),
-          'hist_seq_emb': tf.concat(hist_emb_features, axis=2),
-          'hist_seq_len': hist_seqlen_features[0]
-      }
-    else:
-      features = {
-          'key': key_features[0],
-          'hist_seq_emb': hist_emb_features[0],
-          'hist_seq_len': hist_seqlen_features[0]
-      }
+    hist_tensors = []
+    for hist_seq in hist_seqs:
```

Comment:
`for x in feature_dict.seq_att_map` for forward compatbility

Reply:
done